### PR TITLE
Do not reexport code of imported libraries (e.g. vscode-languageclient)

### PR DESCRIPTION
**Pull request by kaisalmen**
This PR ensure that **monaco-languageclient** and **vscode-ws-jsonrpc** no longer export imported packages.

Additional changes:
- The **browser-lsp** example is replaced with langium statemachine example. No more copied code from Microsoft here.
  - It also makes use of multiple services from `monaco-vscode-api`
- Examples have been moved to one packages with the exception of angular due to its own build approach
- There now is a global `watch` available that watches code in all packages (except for angular and verify where this strategy cannot be applied).
- The usual dependency updates
- node 18 / npm 9 is now configured for Volta
- I am preparing **vitest** inclusion (if you wonder why there now is `tsconfig.src.json`)

I will release next packages to test the implementation and make this final once I think the PR is ready!

### CODE REVIEW

**Comment by kaisalmen** [comment]
@CGNonofr This is not complete, but the new example may be of interest for you. Are the vscode default themes available via some npm package?

**Comment by CGNonofr** [comment]
> @CGNonofr This is not complete, but the new example may be of interest for you.

Interesting! I'll have a look

> Are the vscode default themes available via some npm package? Do you know this by any chance?

Unfortunately no... there are only available as a [vscode extension](https://github.com/microsoft/vscode/tree/main/extensions/theme-defaults). I had to [download the files by hands](https://github.com/CodinGame/monaco-editor-wrapper/blob/246808e506ff230b86d27063c5a428d0b2e7986a/update-extensions.ts#L24) (So I guess the answer is yes are they are available in our [monaco editor wrapper](https://github.com/CodinGame/monaco-editor-wrapper)? :)

File: packages/examples/main/src/langium/main.ts
Comment (CGNonofr) [inline packages/examples/main/src/langium/main.ts:119]: Is there a reason to create this model?

Comment (kaisalmen) [reply packages/examples/main/src/langium/main.ts:119]: Is there a better / preferred way to apply configuration? My aim was to set the theme. The other things were just for playing around.

Comment (CGNonofr) [reply packages/examples/main/src/langium/main.ts:119]: you can just do
```typescript
updateUserConfiguration(`{
    "workbench.colorTheme": "Default Dark+",
    "editor.fontSize": 16,
    "editor.guides.bracketPairs": true
}`)
```

btw to set the theme, you can just use `monaco.editor.setTheme`

Comment (kaisalmen) [reply packages/examples/main/src/langium/main.ts:119]: Thanks 👍

**Comment by kaisalmen** [comment]
Fixed some things and added a node script for fetching themes from code's GH repo.

**Comment by kaisalmen** [comment]
@CGNonofr this is now ready for review.
I have removed the `disposable.ts` from `monaco-languageclient` and only left it in `vscode-ws-jsonrpc`.
Documentation (README has been updated)
I have integrated vscode themes helper into `monaco-languageclient` as a separate export. Maybe this can be even moved to `monaco-vscode-api`

**Comment by kaisalmen** [comment]
@CGNonofr This PR exploded in size. Sorry.
Combing our examples (with exception of angular, webpack and vite verification) reduces maintenance effort.
Do you have time to look at this?

**Comment by CGNonofr** [comment]
> @CGNonofr This PR exploded in size. Sorry. Combing our examples (with exception of angular, webpack and vite verification) reduces maintenance effort. Do you have time to look at this?

I have a deal then: will you have time to handle to update of monaco-vscode-api to monaco@0.36?

**Review by CGNonofr — APPROVED** [review]
hard to review indeed!

LGTM, I just don't know if integrating theme management in monaco-languageclient is relevant

**Comment by kaisalmen** [comment]
> I have a deal then: will you have time to handle to update of monaco-vscode-api to monaco@0.36?

I actually would like to understand how you do this whole transformation. My time next week is very limited, though.

> LGTM, I just don't know if integrating theme management in monaco-languageclient is relevant

I will make up my mind again. I will eventually move it to our monaco-editor-wrapper and have some stripped down version here. Do you think it makes sense to have a "theme helper" in `monaco-vscode-api`?

**Comment by CGNonofr** [comment]
> I actually would like to understand how you do this whole transformation. My time next week is very limited, though.

pretty straighforward:
1 - update monaco-editor and config.vscode.version in package.json, it will download the proper vscode version
2 - fix compilation errors (by updating the code and vscode.patch if required) 
3 - test it and fix runtime error (probably caused by a missing polyfill)

> I will make up my mind again. I will eventually move it to our monaco-editor-wrapper and have some stripped down version here. Do you think it makes sense to have a "theme helper" in monaco-vscode-api?

Maybe we can have something more generic which allows to load a vscode extension instead from a github url?

**Comment by kaisalmen** [comment]
> Maybe we can have something more generic which allows to load a vscode extension instead from a github url?

Sounds like a good idea. Load and use (does `monac-vscode-api` support or are you already doing it elsewhere?) or load and unpack and use what's needed.

**Comment by kaisalmen** [comment]
Btw, I have published a couple of pre-release of 5.0.0 so far. I don't see the v5 immediately after this merge. We should wait for next `monaco-vscode-api` 0.36.0 IMO.

**Comment by CGNonofr** [comment]
> Sounds like a good idea. Load and use (does `monac-vscode-api` support or are you already doing it elsewhere?) or load and unpack and use what's needed.

That basically what we do in our @codingame/monaco-editor-wrapper. It probably make sense to make it more generic and put in inside monaco-vscode-api.

**Comment by kaisalmen** [comment]
> It probably make sense to make it more generic and put in inside monaco-vscode-api.

This sounds like a good idea and could open a couple of more use cases 🎉

**Pull request merged**
